lib/routes/movie_list.dart

```diff
@@ -0,0 +1,68 @@
+import 'package:flutter/material.dart';
+import '/data_classes/movie.dart';
+import '/custom_widgets/movie_preview.dart';
+import '/repositories/i_movie_repository.dart';
+
+class MovieList extends StatelessWidget {
+  final MovieRepository movieRepository = MoviesFromJson();
+  final GenresRepository genresRepository = GenresFromJson();
+  late Future<List<Movie>> movies;
+
+  Future<List<MovieUI>> getMovies() async {
+    List<Movie> movies = await movieRepository.getMovies();
+    Map<int, String> genres = await genresRepository.getFilms();
+    return movies
+        .map((e) => MovieUI(e, e.genres.map((e) => genres[e]).toList()))
+        .toList();
+  }
+
+  MovieList({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return FutureBuilder(
+        future: getMovies(),
```

Comment:
We can add some Flutter best practices to the Future handling in the UI. Please read this:
https://api.flutter.dev/flutter/widgets/FutureBuilder-class.html

> If the future is created at the same time as the FutureBuilder, then every time the FutureBuilder's parent is rebuilt, the asynchronous task will be restarted. A general guideline is to assume that every build method could get called every frame, and to treat omitted calls as an optimization.

TLDR, you could make this widget Stateful, declare a Future<T> movies at the beginning, init that future in the initState method, and use that future in the FutureBuilder. Like here:
https://stackoverflow.com/a/68966640

Reply:
Done